File: Java/org/oobench/patterns/flyweight/Screen.java
package org.oobench.patterns.flyweight;

public class Screen {

    private int size;
    private byte[] screen;

    public Screen(int theSize) {
        size = theSize;
        screen = new byte[size];
    }

    public int getSize() {
        return size;
    }

    public byte[] getScreenBuffer() {
        return screen;
    }
}

File: C++/Patterns/Flyweight/Screen.cpp
#include "Screen.hpp"

namespace oobench {

    Screen::Screen(long theSize) : size(theSize) {
        screen = new char[size];
    }

    Screen::~Screen() {
        delete[] screen;
    }

    long Screen::getSize() const {
        return size;
    }

    char* Screen::getScreenPointer() const {
        return screen;
    }
}
